### DISCUSSION TITLE
Cannot change font size of axis tick labels

### DISCUSSION THREAD

**Discussion by ccastelblanco** [pyvista/pyvista#4721] [Q&A]
Hi,

I am wondering how I can change the font size of tick labels, as the font_size parameter of the plotter show_grid function is not working for me.

Here, I attach images of the issue on a plot I did:

![image](https://github.com/pyvista/pyvista/assets/97355202/49ef79e8-7f65-44b2-aad6-942a3ae74cd5)

![image](https://github.com/pyvista/pyvista/assets/97355202/1e1d4ec3-ee60-439b-bc9c-f800cf296f60)
 
I appreciate your kind help.

Regards, 

Carlos

**Comment by tkoyama010** [comment]
@ccastelblanco
Hello! The `font_size` seemed to work in my environment.
```python
import pyvista as pv
from pyvista import examples

mesh = examples.load_random_hills()

plotter = pv.Plotter()
actor = plotter.add_mesh(
    mesh, cmap='terrain', show_scalar_bar=False
)
actor = plotter.show_grid(
    color='gray',
    location='outer',
    grid='back',
    ticks='outside',
    xtitle='Longtitude (°)',
    ytitle='Latitude (°)',
    ztitle='Elevation (km)',
    font_size=10,
)
plotter.show()
```
![image](https://github.com/pyvista/pyvista/assets/7513610/97216a72-b4e8-473f-8a3c-ebf3035f83c3)
```python
import pyvista as pv
from pyvista import examples

mesh = examples.load_random_hills()

plotter = pv.Plotter()
actor = plotter.add_mesh(
    mesh, cmap='terrain', show_scalar_bar=False
)
actor = plotter.show_grid(
    color='gray',
    location='outer',
    grid='back',
    ticks='outside',
    xtitle='Longtitude (°)',
    ytitle='Latitude (°)',
    ztitle='Elevation (km)',
    font_size=20,
)
plotter.show()
```
![image](https://github.com/pyvista/pyvista/assets/7513610/9074435a-1968-46b7-8641-06a6b885a000)

To reproduce your error, could you run the following code and share your environment with us?
If you can also share the data you are plotting, we are more likely to success the reproduce.

```python
# Get system info
import pyvista as pv
print(pv.Report())
```
```
--------------------------------------------------------------------------------
  Date: Fri Jul 28 19:55:36 2023 東京 (標準時)

                OS : Windows
            CPU(s) : 16
           Machine : AMD64
      Architecture : 64bit
               RAM : 51.9 GiB
       Environment : IPython
        GPU Vendor : Intel
      GPU Renderer : Intel(R) UHD Graphics 630
       GPU Version : 4.5.0 - Build 30.0.100.9805
  MathText Support : False

  Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64
  bit (AMD64)]

           pyvista : 0.41.1
               vtk : 9.2.6
             numpy : 1.24.2
        matplotlib : 3.7.1
            scooby : 0.7.1
             pooch : v1.7.0
           imageio : 2.27.0
           IPython : 8.11.0
          colorcet : 3.0.1
           cmocean : v3.0.3
        ipyvtklink : Trouble importing
        ipywidgets : 8.0.6
             scipy : 1.10.1
              tqdm : 4.65.0
            meshio : 5.3.4
        jupyterlab : 3.6.3
         pythreejs : 2.4.2
    pytest_pyvista : 0.1.7
             trame : 2.4.0
      trame_client : 2.7.5
      trame_server : 2.11.0
         trame_vtk : 2.4.4
      nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

**Reply by ccastelblanco** [reply]
Hello Tetsuo

I ran the code of the example and I could change the font size without problems. Then, it seems like something happens when I use my data.

On the other hand, after running the code for the system info, this is what is displayed:

--------------------------------------------------------------------------------
   Date: Tue Aug 01 15:00:22 2023 Hora est. Pacífico, Sudamérica

                OS : Windows
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
               RAM : 24.0 GiB
       Environment : Jupyter
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : GeForce 210/PCIe/SSE2
       GPU Version : 3.3.0  
      MathText Support : False

      Python 3.10.0 | packaged by conda-forge | (default, Nov 20 2021, 02:18:13)
      [MSC v.1916 64 bit (AMD64)]

           pyvista : 0.41.1
               vtk : 9.2.6
             numpy : 1.25.1
        matplotlib : 3.7.2
            scooby : 0.7.2
             pooch : v1.7.0
           imageio : 2.31.1
             PyQt5 : 5.15.7
           IPython : 8.14.0
        ipyvtklink : 0.2.2
        ipywidgets : 8.0.7
             scipy : 1.11.1
        jupyterlab : 4.0.3
         pythreejs : 2.4.2
             trame : 3.0.2
      trame_client : 2.10.0
      trame_server : 2.11.7
         trame_vtk : 2.5.8
      nest_asyncio : 1.5.6
--------------------------------------------------------------------------------

As requested, here I send the files of the plot:

[test_files.zip](https://github.com/pyvista/pyvista/files/12233726/test_files.zip)


Thank you!

**Reply by tkoyama010** [reply]
Sorry for the delay in replying. I was able to reproduce the font not changing at hand. Please wait a little longer.

**Reply by tkoyama010** [reply]
@ccastelblanco  I found the solution. This is happening because the scale is set after the font is set. If you swap the order of the settings you will get the result you are looking for.
<details>
<summary><b>Code</b></summary>

```python
import gemgis as gg
import pyvista as pv
pv.set_jupyter_backend('static')
import numpy as np
import pandas as pd

pv.global_theme.font.family = 'arial'
pv.global_theme.font.color = 'grey'

topo = gg.visualization.read_raster(path="dem.tif",
                                    nodata_val=9999.0,
                                    name='Elevation [m]')
topo = topo.warp_by_scalar(scalars="Elevation [m]", factor=0.001)
df = pd.read_csv("data.csv")

values = np.reshape(df['value'], (13,17,9)).swapaxes(0,1)

xg = np.reshape(df['x'], (13,17,9)).swapaxes(0,1) 
yg = np.reshape(df['y'], (13,17,9)).swapaxes(0,1) 
zg = np.reshape(-df['z'], (13,17,9)).swapaxes(0,1) 

cell_interv = 10/111.1
dx = (xg[1,1,1]-xg[0,0,0])/2
dy = (yg[1,1,1]-yg[0,0,0])/2
dz = (zg[1,1,1]-zg[0,0,0])/2

grid = pv.ImageData()
grid.dimensions = np.array(values.shape) + 1
grid.origin = (-74.4-dx, 6.7-dy, -20+dz)
grid.spacing = (cell_interv, cell_interv, cell_interv/4*111.1)
grid['value'] = values.ravel(order='F')

grid_map_slice = grid.slice(normal='z', origin=(grid.origin[0], grid.origin[1], 0))

origin = (-73.518175, 7.184844999999999, 0)
angle = -0.591679574121426
normal = np.array([-np.sin(angle), np.cos(angle), 0.0]).dot(np.pi / 2.0) 
prof_slice = grid.slice(normal= normal, origin=origin)

topo_clip = topo.clip(normal= -normal, origin=origin)
grid_map_slice_clip = grid_map_slice.clip(normal= -normal, origin=origin)

p = pv.Plotter()
sargs = dict(fmt="%.0f", color='grey')
p.add_mesh(mesh=topo_clip, cmap='Greys', scalar_bar_args=sargs, opacity=0.7, show_scalar_bar=False)
p.add_mesh(grid_map_slice_clip, opacity=0.8, interpolate_before_map=True)
p.add_mesh(prof_slice, 
           opacity=0.9,
           )
p.set_scale(zscale=0.01) # Set scale before set the font of grid.  
p.show_grid(color='grey', 
            location='outer', 
            grid='back', 
            ticks='outside',
            xtitle='Longitude (°)', 
            ytitle='Latitude (°)', 
            ztitle='Elevation (km)',
            font_size=10,
            fmt='%.1f'
            )
p.set_background(color='white')
p.camera_position = 'xz'
p.camera.elevation = 30
p.camera.azimuth = -30
p.show()
```

</details>

![image](https://github.com/pyvista/pyvista/assets/7513610/34a9eca7-0099-413c-ae13-8dc97791f263)

**Reply by tkoyama010** [reply]
Same with https://github.com/pyvista/pyvista/issues/4768 .

**Reply by AlexanderJuestel** [reply]
A didn't see this @tkoyama010, thanks for referencing it!

**Reply by AlexanderJuestel** [reply]
I have attached a little notebook which is more related to #4768. Once you set a scale other than 1 for the Z axis, you can no longer change the font size. If the scale comes before the bounds, then you get small labels. If the scale comes after the bounds, then you get large labels. 

I wonder if this is intended :) 

[PyVista_Scale_Font_Size.zip](https://github.com/pyvista/pyvista/files/12344898/PyVista_Scale_Font_Size.zip)

**Reply by tkoyama010** [reply]
Please see the reply in https://github.com/pyvista/pyvista/issues/4768.

**Reply by ccastelblanco** [reply]
Thank you for your help! 

The axis tick labels with this got smaller, but it is not possible to change them to a different font size. Additionally, I see that now z axis ticks disappeared.

What could you suggest?
